src/Lykke.Service.PayInvoicePortal/Controllers/Api/User/SettingsController.cs

```diff
@@ -0,0 +1,195 @@
+
+                    availableBaseAssets.TryAdd(baseAssetId, asset?.DisplayId ?? baseAssetId);
+                }
+
+                var settings = new SettingsResponse
+                {
+                    MerchantDisplayName = merchant.DisplayName,
+                    EmployeeFullname = User.GetName(),
+                    EmployeeEmail = User.GetEmail(),
+                    AvailableBaseAssets = availableBaseAssets.Select(o => new AssetItemViewModel(o.Key, o.Value)).ToList().OrderBy(_ => _.Title),
+                    BaseAssetId = baseAssetId,
+                    MerchantId = merchantId,
+                    MerchantApiKey = merchant.ApiKey,
+                    HasPublicKey = !string.IsNullOrEmpty(publicKeyInfo.RsaPublicKey)
+                };
+
+                return Ok(settings);
+            }
+            catch (Exception e)
+            {
+                _log.Error(e, new { employeeId, merchantId }.ToJson());
+
+                return BadRequest(ErrorResponse.Create(PayInvoicePortalApiErrorCodes.UnexpectedError));
+            }
+        }
+
+        [HttpPost]
+        [Route("generateRsaKeys")]
+        [ProducesResponseType(typeof(RsaPrivateKeyResponse), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> GenerateRsaKeys()
+        {
+            var merchantId = User.GetMerchantId();
+
+            try
+            {
+                var rsaKeysResponse = await _payAuthClient.GenerateRsaKeysAsync(new GenerateRsaKeysRequest
+                {
+                    ClientId = merchantId,
+                    ClientDisplayName = await _merchantService.GetMerchantNameAsync(merchantId)
+                });
+
+                return Ok(new RsaPrivateKeyResponse
+                {
+                    RsaPrivateKey = rsaKeysResponse.PrivateKey
+                });
+            }
+            catch (Exception e)
+            {
+                _log.Error(e, new { merchantId }.ToJson());
+
+                return BadRequest(ErrorResponse.Create(PayInvoicePortalApiErrorCodes.UnexpectedError));
+            }
+        }
+
+        [HttpPost]
+        [Route("baseAsset")]
+        [ValidateModel]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> SetBaseAsset([FromBody] SetBaseAssetModel model)
+        {
+            var merchantId = User.GetMerchantId();
+
+            try
+            {
+                await _payInvoiceClient.SetMerchantSettingAsync(new MerchantSetting
+                {
+                    MerchantId = merchantId,
+                    BaseAsset = model.BaseAssetId
+                });
+
+                return Ok();
+            }
+            catch (Exception e)
+            {
+                _log.Error(e, new { merchantId, model.BaseAssetId }.ToJson());
```

Comment:
As per payInvoice implementation you can get here ErroResponseException in case of NotFound and BadRequest response statuses. I wouldn't reflect this in ErrorResponse as unexpected error.

Reply:
there are two sides:
From perspective of api we should log such errors, and from perspective of User in UI - it is unexpected behavior and in this case the User should contact support.